src/commands/subcommands/labels.ts

```diff
@@ -0,0 +1,81 @@
+import { HttpStatusCode, IModify } from '@rocket.chat/apps-engine/definition/accessors';
+import { SlashCommandContext } from '@rocket.chat/apps-engine/definition/slashcommands';
+import { LayoutBlock } from '@rocket.chat/ui-kit';
+
+import { TodoistApp } from '../../../TodoistApp';
+import { MiscEnum } from '../../enums/Misc';
+import {
+  getActionsBlock,
+  getButton,
+  getContextBlock,
+  getSectionBlock,
+} from '../../helpers/blockBuilder';
+import { getLabelsUrl } from '../../helpers/const';
+import { ILabel } from '../../interfaces/labels';
+
+export async function labels(
+  app: TodoistApp,
+  modify: IModify,
+  context: SlashCommandContext
+): Promise<void> {
+  const user = context.getSender();
+  const room = context.getRoom();
+
+  const response = await app.getHttpHelperInstance().get(user, getLabelsUrl());
+
+  if (response.statusCode !== HttpStatusCode.OK) {
+    const msg = modify
+      .getCreator()
+      .startMessage()
+      .setText(`❗️ Unable to retrieve labels! \n Error ${JSON.stringify(response)}`)
+      .setRoom(room);
+    await modify.getNotifier().notifyUser(user, msg.getMessage());
+    return;
+  }
+
+  if (!response.data || response.data.length === 0) {
+    const msg = modify
+      .getCreator()
+      .startMessage()
+      .setText('No personal labels found. Create one using the Todoist app or website.')
+      .setRoom(room);
+    await modify.getNotifier().notifyUser(user, msg.getMessage());
+    return;
+  }
+
+  const builder = modify.getCreator().startMessage().setRoom(room);
+  const blocks = (await Promise.all(response.data.map(createLabelSection))).reduce(
+    (acc, val) => acc.concat(val),
+    []
+  ) as LayoutBlock[];
+  builder.setBlocks(blocks);
+  await modify.getNotifier().notifyUser(user, builder.getMessage());
+}
+
+async function createLabelSection(label: ILabel): Promise<LayoutBlock[]> {
+  const labelNameBlock = getSectionBlock(label.name);
+  const labelContextBlock = getContextBlock(
+    `Color: ${String(label.color).charAt(0).toUpperCase() + String(label.color).slice(1)} | ` +
```

Comment:
Lets put this into a separate function wdyt?
this function is dealing contextually with label blocks, I don't think it is good for it to know exactly how the color field is internally